app/core/service/course/course.service.js

```diff
@@ -1,40 +1,51 @@
-(function () {
-    'use strict';
+'use strict';
 
-    angular
-        .module('course')
-        .factory('courseService', ['$http', function($http) {
-            function loadCourses() {
-                return $http.get('data/courses.json');
-            }
+angular
+    .module('course')
+    .factory('courseService', [ '$http', function($http) {
+        function loadCourses() {
+            return $http.get('data/courses.json');
+        }
+        function addCourse(data, array) {
+            array.unshift(data);
+        }
+        function editCourse(data, array) {
+            var course = {
+                title: data.title,
+                description: data.description,
+                uploadDate: data.uploadDate
+            };
+            this.addDisplayDateAndTimeAfterUpdating(course);
```

Comment:
It looks strange for me. Are you sure that it works in the same way as it's expected?

Reply:
removed